i2c/bmp280_i2c/bmp280_i2c.c

```diff
@@ -0,0 +1,254 @@
+  i2c_write_blocking(i2c_default, (addr & BMP280_WRITE_MODE), buf, 2, false);
+
+  // osrs_t x1, osrs_p x4, normal mode operation
+  const uint8_t reg_ctrl_meas_val = (0x01 << 5) | (0x03 << 2) | (0x03);
+  buf[0] = REG_CTRL_MEAS;
+  buf[1] = reg_ctrl_meas_val;
+  i2c_write_blocking(i2c_default, (addr & BMP280_WRITE_MODE), buf, 2, false);
+}
+
+void bmp280_read_raw(int32_t* temp, int32_t* pressure) {
+  // BMP280 data registers are auto-incrementing and we have 3 temperature and
+  // pressure registers each, so we start at 0xF7 and read 6 bytes to 0xFC note:
+  // normal mode does not require further ctrl_meas and config register writes
+
+  uint8_t buf[6];
+  i2c_write_blocking(i2c_default, (addr & BMP280_WRITE_MODE), &REG_PRESSURE_MSB, 1, true);  // true to keep master control of bus
+  i2c_read_blocking(i2c_default, (addr & BMP280_READ_MODE), buf, 6, false);  // false - finished with bus
+
+  // store the 20 bit read in a 32 bit signed integer for conversion
+  *pressure = (buf[0] << 12) | (buf[1] << 4) | (buf[2] >> 4);
+  *temp = (buf[3] << 12) | (buf[4] << 4) | (buf[5] >> 4);
+}
+
+void bmp280_reset() {
+  // reset the device with the power-on-reset procedure
+  uint8_t buf[2] = { REG_RESET, 0xB6 };
+  i2c_write_blocking(i2c_default, (addr & BMP280_WRITE_MODE), buf, 2, false);
+}
+
+int32_t bmp280_convert(int32_t temp, struct bmp280_calib_param* params) {
+  // intermediate function that calculates the fine resolution temperature
+  // used for both pressure and temperature conversions
+  int32_t var1, var2;
+  var1 = ((((temp >> 3) - ((int32_t)params->dig_t1 << 1))) * ((int32_t)params->dig_t2)) >> 11;
+  var2 = (((((temp >> 4) - ((int32_t)params->dig_t1)) * ((temp >> 4) - ((int32_t)params->dig_t1))) >> 12) * ((int32_t)params->dig_t3)) >> 14;
+  return var1 + var2;
+}
+
+int32_t bmp280_convert_temp(int32_t temp, struct bmp280_calib_param* params) {
+  // use the 32-bit fixed point compensation implementation given in the
+  // datasheet
+  int32_t t_fine = bmp280_convert(temp, params);
+  return (t_fine * 5 + 128) >> 8;
+}
+
+int32_t bmp280_convert_pressure(int32_t pressure, int32_t temp, struct bmp280_calib_param* params) {
+  int32_t t_fine = bmp280_convert(temp, params);
+
+  int32_t var1, var2;
+  uint32_t converted = 0.0;
+  var1 = (((int32_t)t_fine) >> 1) - (int32_t)64000;
+  var2 = (((var1 >> 2) * (var1 >> 2)) >> 11) * ((int32_t)params->dig_p6);
+  var2 += ((var1 * ((int32_t)params->dig_p5)) << 1);
+  var2 = (var2 >> 2) + (((int32_t)params->dig_p4) << 16);
+  var1 = (((params->dig_p3 * (((var1 >> 2) * (var1 >> 2)) >> 13)) >> 3) + ((((int32_t)params->dig_p2) * var1) >> 1)) >> 18;
+  var1 = ((((32768 + var1)) * ((int32_t)params->dig_p1)) >> 15);
+  if (var1 == 0) {
+    return 0;  // avoid exception caused by division by zero
+  }
+  converted =
+    (((uint32_t)(((int32_t)1048576) - pressure) - (var2 >> 12))) * 3125;
+  if (converted < 0x80000000) {
+    converted = (converted << 1) / ((uint32_t)var1);
+  }
+  else {
```

Comment:
ping @kilograham What bracket-formatting-style should be used here?

Reply:
`} else {`